test/catalog/catalog_test.cpp

```diff
@@ -502,4 +502,64 @@ TEST_F(CatalogTests, GetIndexObjectsTest) {
   txn_manager_->Commit(txn, transaction::TransactionUtil::EmptyCallback, nullptr);
 }
 
+/*
+ * Exercise a bunch of scenarios of the DDL lock semantics
+ */
+// NOLINTNEXTLINE
+TEST_F(CatalogTests, DDLLockTest) {
+  // txn2 is used to verify that older txns can't acquire the lock
+  auto *txn2 = txn_manager_->BeginTransaction();
```

Comment:
nit: It's easier to understand the tested ordering if the the number in `txn#` corresponds to the BEGIN ordering since it will at least be consistent with the begin timestamp ordering.